### PR TITLE
update and merge quickstart notebooks

### LINKED ISSUE #229: Move examples from Readme into docs.
We have a number of examples in the Readme, but we should move them to ReadTheDocs.

### LINKED ISSUE #663: Merge the two quickstart notebooks?
Currently we have two quickstart notebooks:

* Learn Optax: https://optax.readthedocs.io/en/latest/optax-101.html , featured from the "Getting started" section
* Quickstart with Optax: https://optax.readthedocs.io/en/latest/_collections/examples/quick_start.html , linked only from the Example gallery (but not from the Getting started section).

Since both notebooks are not too long, I think it would be a good idea to merge both, or at least put them together in the same section.

CC @holounic @mtthss @rosshemsley : curious to get your thoughts on this one

### PR DESCRIPTION
**Pull request by amosyou**
updates `optax101.ipynb` notebook with README examples (same as #435, closes #229) and merges `optax101.ipynb` with `quick_start.ipynb` (closes #663)

### CODE REVIEW

**Comment by vroulet** [comment]
Hello @amosyou,
And thanks again! As far as I can see, the optax-101 is just the README file put in notebook format. 
@fabianp had made a pass on the README file recently to correct diverse typos. Could you sync with main and use the newest version of the README?
(examples of typos are broken links as we changed the docs structure, you may not want to redo that :) )

**Comment by amosyou** [comment]
sounds good! also wanted to take note of #663, @fabianp did we come to a consensus on whether to merge the notebooks? if so, happy to include those changes here in addition to adjusting for the typos

**Comment by fabianp** [comment]
yes, ideally we would merge both notebooks. What you propose sounds great @amosyou !

**Comment by amosyou** [comment]
@fabianp I decided to include both examples in the notebook (ie. linear model from quick_start, mlp from optax-101). Coming from someone who's new to optax, I found the linear model example really useful and easy to understand, but the mlp example is good for understanding the params as a dict. Let me know what you think!

**Comment by fabianp** [comment]
sounds perfect. Let me know when I should take a look at the diff 😄

**Comment by amosyou** [comment]
yup feel free to take a look!

File: docs/optax-101.ipynb
Comment (fabianp) [reply docs/optax-101.ipynb:1]: Instead of having a hard-coded link such as `https://optax.readthedocs.io/en/latest/api/common_losses.html`, use [sphinx references](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#reference-roles) whenever possible.

So for example, instead of 
```The loss function (L2 Loss) comes from optax's [common loss functions](https://optax.readthedocs.io/en/latest/api/common_losses.html) via `optax.l2_loss()`.```
I would use ``` The loss function (L2 Loss) comes from optax's {doc}`losses <api/losses>` via {ref}`optax.l2_loss`.```

It's a bit uglier to read but the generated webpage is much nicer 😄

Comment (fabianp) [reply docs/optax-101.ipynb:1]: The cell below `### Gradient Transformations` raises an exception `NameError: name 'init' is not defined`, please fix that

**Review by fabianp — CHANGES_REQUESTED** [review]
A few cosmetic comments, thanks @amosyou !

**Comment by fabianp** [comment]
hey @amosyou , do you think you'll have cycles to implement the changes in the next couple of days? and if not, do you mind if I take over? I would really like to merge this before making any other changes to the documentation

**Comment by amosyou** [comment]
yup! apologies for the delay, I can get the changes in today

**Comment by fabianp** [comment]
thanks! And no rush, I just want this to be merged before I tackle other
doc improvements

File: docs/optax-101.ipynb
Comment (amosyou) [reply docs/optax-101.ipynb:1]: the cell below `### Gradient Transformations` technically isn't meant to be run, it's pseudocode for the reader to understand the init/update functions. I changed it back to inline code snippet, would proceeding like this look better?

Comment (amosyou) [reply docs/optax-101.ipynb:1]: For some reason the code cell under `### Gradient Transformation` is facing an error, but the code cell under `### Applying updates` is similar but no errors are found there

**Review by fabianp — APPROVED** [review]
Thanks @amosyou ! This document being the entry point for most users, it's an extremely important contribution that makes a big difference. Thanks again!

**Comment by amosyou** [comment]
np! I had a couple comments under your code review, there might still be an error under one of the code cells. Also, I could either do it here or open a new issue to delete the other notebook in the gallery now that it's been combined

**Comment by fabianp** [comment]
I've seen, no worries. I'll make a pass on it before merging :-)

**Pull request merged**
